vcf.c

```diff
@@ -3641,13 +3641,11 @@ bcf_hdr_t *bcf_hdr_merge(bcf_hdr_t *dst, const bcf_hdr_t *src)
                 {
                     hts_log_warning("Trying to combine \"%s\" tag definitions of different lengths",
                         src->hrec[i]->vals[0]);
-                    ret |= 1;
```

Comment:
I get the feeling that there was an intention to do something else if this problem (or the one below) was detected.  While it has been ignored for now, I think it would be good to check while our attention has been drawn to this bit of code.  Maybe @pd3 could let us know how catastrophic the problems being warned about might be?

Reply:
They are all like that.  Things that did something, or were meant to do something, and have been left behind.  If the people who wrote it want an alternative fix then they are free to overwrite my changes.

Reply:
In this case I don't think it ever did anything.  But it would be good to at least raise an issue if we think it really should be doing something, so we remember to fix it.

Reply:
It may be a problem but does not have to be. Probably best if the function was able to set an error code which could be optionally checked by the caller?

Reply:
Unfortunately the only way the function can report errors at the moment is to return NULL.  While it already does for other detected problems, it never has for these.  As in the case of these particular warnings you'll always have supplied a `dst` you could recover from a NULL return, but it would be a bit of a faff.

Presumably if you get these warnings, one of the `dst` headers has the wrong type.  What would be the consequence of trying to use it, and is it bad enough to merit returning an error from this function?

Reply:
Sorry for not being clear, I meant if maybe `errno` could be (mis)used for this.

The consequence depends on the user application. The current uses in BCFtools are okay with it, another check can be  performed at the data record level.

Probably best and easiest would be to provide a new function, in case it is ever needed. For now I am completely okay with leaving it as is.

Reply:
You'd have to return `NULL` as well as setting `errno`, otherwise the caller wouldn't know to check the latter (following the convention that `errno` is left unchanged if everything worked).

If you can check this on data records then I guess it's OK to ignore for now.